src/main/java/com/microsoft/aad/msal4j/RefreshTokenRequest.java

```diff
@@ -6,18 +6,54 @@
 import com.nimbusds.oauth2.sdk.RefreshTokenGrant;
 import com.nimbusds.oauth2.sdk.token.RefreshToken;
 
+import java.util.Set;
+import java.util.TreeSet;
+
 class RefreshTokenRequest extends MsalRequest {
 
+    private SilentRequest parentSilentRequest;
+    private RefreshTokenParameters parameters;
+
     RefreshTokenRequest(RefreshTokenParameters parameters,
-                        ClientApplicationBase application,
-                        RequestContext requestContext){
+                        AbstractClientApplicationBase application,
+                        RequestContext requestContext) {
         super(application, createAuthenticationGrant(parameters), requestContext);
+        this.parameters = parameters;
+    }
+
+    RefreshTokenRequest(RefreshTokenParameters parameters,
+                        AbstractClientApplicationBase application,
+                        RequestContext requestContext,
+                        SilentRequest silentRequest) {
+        this(parameters, application, requestContext);
+        this.parentSilentRequest = silentRequest;
     }
 
     private static AbstractMsalAuthorizationGrant createAuthenticationGrant(
-            RefreshTokenParameters parameters){
+            RefreshTokenParameters parameters) {
 
         RefreshTokenGrant refreshTokenGrant = new RefreshTokenGrant(new RefreshToken(parameters.refreshToken()));
         return new OAuthAuthorizationGrant(refreshTokenGrant, parameters.scopes());
     }
+
+    private String getRefreshTokenRequestFullThumbprint() {
+        String DELIMITER = ".";
```

Comment:
This delimeter is used in multiple places. Might make sense to add to Contants file.

Reply:
ok